src/Task/MapRoute.pm

```diff
@@ -271,8 +273,66 @@ sub iterate {
 
 	} elsif ( $portals_lut{"$self->{mapSolution}[0]{map} $self->{mapSolution}[0]{pos}{x} $self->{mapSolution}[0]{pos}{y}"}{source} ) {
 		# This is a portal solution
-
-		if ( distance($self->{actor}{pos_to}, $self->{mapSolution}[0]{pos}) < 2 ) {
+		
+		if ($self->{missing_portal}) {
+		
+			if (!$config{route_tryToGuessMissingPortalByDistance} || $portalsList->size == 0) {
```

Comment:
Aha, so there's a second copy of this code inside the `route_tryToGuessMissingPortalByDistance` branch. Maybe we don't need the `$portalsList->size == 0` check here at all, then?